packages/pyright-internal/src/backgroundThreadBase.ts

```diff
@@ -6,31 +6,36 @@
  * base class for background worker thread.
  */
 
-import { MessagePort, parentPort } from 'worker_threads';
-
 import { OperationCanceledException, setCancellationFolderName } from './common/cancellationUtils';
 import { ConfigOptions } from './common/configOptions';
 import { LogLevel } from './common/console';
 import * as debug from './common/debug';
 import { FileSystem } from './common/fileSystem';
-import { FileSpec } from './common/pathUtils';
-import { createFromRealFileSystem } from './common/realFileSystem';
+import { FileSpec, normalizeSlashes } from './common/pathUtils';
+import { UnifiedPort } from './common/workers';
 import { PyrightFileSystem } from './pyrightFileSystem';
+import { TestFileSystem } from './tests/harness/vfs/filesystem';
 
 export class BackgroundThreadBase {
     protected fs: FileSystem;
+    // Ugh.
+    protected _realFs: FileSystem;
 
-    protected constructor(data: InitializationData) {
+    constructor(protected parentPort: UnifiedPort | null, data: InitializationData) {
         setCancellationFolderName(data.cancellationFolderName);
 
         // Stash the base directory into a global variable.
         (global as any).__rootDirectory = data.rootDirectory;
 
-        this.fs = new PyrightFileSystem(createFromRealFileSystem(this.getConsole()));
+        this._realFs = new TestFileSystem(false, {
+            cwd: normalizeSlashes('/'),
+        });
+        // Browser hack. We need to parameterize this.
+        this.fs = new PyrightFileSystem(this._realFs);
     }
```

Comment:
Other than testing the NodeJS case, this is the only substantive issue remaining.

Reply:
I don't think this PR should try to remove the use of the test file system, so let's just move the hack to browser-only code.

Perhaps the work done for github.dev will result in a file system implementation that uses 
https://code.visualstudio.com/api/references/vscode-api#FileSystem

That's non-trivial though because it's async.

Reply:
Pushed the creation to an abstract method. It's a wart, and I'll revisit, but it's probably enough to be useful.